Utils.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace DUCK.HieriarchyBehaviour
+{
+	internal static class Utils
+	{
+		internal static TBehaviour CloneBehaviour<TBehaviour>(TBehaviour behaviourToClone, GameObject parent)
+			where TBehaviour : MonoBehaviour
+		{
+			var behaviour = Object.Instantiate(behaviourToClone, parent.transform);
+			behaviour.name = behaviourToClone.name;
+			behaviour.transform.localPosition = Vector3.zero;
+			return behaviour;
+		}
+
+		internal static TBehaviour CreateGameObjectWithBehaviour<TBehaviour>(GameObject parent)
+			where TBehaviour : MonoBehaviour
+		{
+			var behaviour = new GameObject(typeof(TBehaviour).Name).AddComponent<TBehaviour>();
+			behaviour.transform.SetParent(parent.transform);
+			behaviour.transform.localPosition = Vector3.zero;
+			return behaviour;
+		}
+
+		internal static TBehaviour InstantiateResource<TBehaviour>(string path, GameObject parent, bool worldPositionStay = true)
+			where TBehaviour : MonoBehaviour
+		{
+			var loadedBehaviour = Resources.Load<TBehaviour>(path);
+			var behaviour = Object.Instantiate(loadedBehaviour, parent.transform, worldPositionStay);
+			behaviour.name = loadedBehaviour.name;
+			return behaviour;
+		}
+
+		internal static void DestroyChild(GameObject parent, MonoBehaviour child)
+		{
+			if (child.transform.parent != parent.transform)
+			{
+				throw new ArgumentException(string.Format("{0} is not a child transform of {1}", child.name, parent.name));
+			}
+
+			if (Application.isEditor)
+			{
+				Object.DestroyImmediate(child.gameObject);
```

Comment:
Unity awkwardly made two the destroy methods, where `Destroy` only works runtime and `DestroyImmediate` only works in the editor.

The reason I need both are for for the Tests but its also useful if for some reason you want to create an editor tool that needs to create children MonoBehaviours and initialize them, then its supported.

Reply:
Yes I know you shouldn't use `DestroyImmediate` in the runtime but the piece that I was missing here is that Destroy doesn't work in the editor **at all** and will actually throw instead.

Frustrating because that means the behaviour of object destruction is different between Editor and the runtime and probably harder to debug.